packages/surveys/src/components/wrappers/StackedCardsContainer.tsx

```diff
@@ -45,35 +45,9 @@ export const StackedCardsContainer = ({
     return survey.questions.findIndex((question) => question.id === currentQuestionId);
   }, [currentQuestionId, survey.welcomeCard.enabled, survey.questions]);
 
-  const [prevQuestionIdx, setPrevQuestionIdx] = useState(questionIdxTemp - 1);
-  const [currentQuestionIdx, setCurrentQuestionIdx] = useState(questionIdxTemp);
-  const [nextQuestionIdx, setNextQuestionIdx] = useState(questionIdxTemp + 1);
-  const [visitedQuestions, setVisitedQuestions] = useState<number[]>([]);
-
-  useEffect(() => {
-    if (questionIdxTemp > currentQuestionIdx) {
-      // Next button is clicked
-      setPrevQuestionIdx(currentQuestionIdx);
-      setCurrentQuestionIdx(questionIdxTemp);
-      setNextQuestionIdx(questionIdxTemp + 1);
-      setVisitedQuestions((prev) => {
-        return [...prev, currentQuestionIdx];
-      });
-    } else if (questionIdxTemp < currentQuestionIdx) {
-      // Back button is clicked
-      setNextQuestionIdx(currentQuestionIdx);
-      setCurrentQuestionIdx(questionIdxTemp);
-      setPrevQuestionIdx(visitedQuestions[visitedQuestions.length - 2]);
-      setVisitedQuestions((prev) => {
-        if (prev.length > 0) {
-          const newStack = prev.slice(0, -1);
-          return newStack;
-        }
-        return prev;
-      });
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [questionIdxTemp]);
```

Comment:
These changes were introduced to fix another bug(https://github.com/formbricks/formbricks/commit/a91c9db4e0c2092e31e767ac771b72acff7925a5). In the case of the logic jump, the destination question was not animating. Can you please try creating a solution without removing this code, or can you fix it while keeping the same behaviour?
The incorrect index thing is something we need to check.

Reply:
Thanks for review!
I missed destination question feature. I reverted this commit.
To address the first reason I mentioned, , I chose to replace the code below with the `resetQuestionProgress` code instead of modifying `StackedCardsContainer` code.
> https://github.com/formbricks/formbricks/blob/e9c5b00628ca4a39fb3f8c9fab76bb9244d8a33d/packages/ui/PreviewSurvey/index.tsx#L201